### PR TITLE
Port of NSIS installer

### PR DESCRIPTION
**Pull request by joedborg**
~~WIP to trigger and debug the Action build.~~

This moves the new NSIS based installer out of a private repo and into MicroK8s.  It also adds the framework to building the installer via GitHub actions.

Needless to say, this should be squashed when merged.

### CODE REVIEW

**Comment by ktsakalozos** [comment]
Merging this now and will review it tomorrow, thank you @joedborg

**Comment by ktsakalozos** [comment]
Here some issues when testing the installer:
- old installations where not cleaned/replaced so even though the installation finished the microk8s.exe in my PATH was the old one and it was failing to contact the MicroK8sVM.
- my multipass but the service was disabled/stopped so even though multipass was detected the installation of the microk8s-vm failed.

File: installer/requirements.txt [hunk 1]
```diff
@@ -11,3 +10,4 @@ toml==0.10.0
 wheel
 pyinstaller
 setuptools<45.0.0  # https://github.com/pypa/setuptools/issues/1963
+https://pyyaml.org/download/pyyaml/PyYAML-5.3.1-cp38-cp38-win_amd64.whl
```
Comment (ktsakalozos) [inline installer/requirements.txt:13]: Could we add the `sys_platform == 'win32'` flag so pyinstaller can produce a binary for linux as well?

File: installer/windows/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+Uses the NSIS package to create the installer.  
+
+### Prerequisites
+
+* Requires [NSIS 3](https://nsis.sourceforge.io/Download).
+* Requires the [EnVar plug in](https://nsis.sourceforge.io/EnVar_plug-in) to compile.
+
+### Building
+
+2 files must be places in the working directory before compiling the installer.  These are:
```
Comment (ktsakalozos) [inline installer/windows/README.md:13]: *placed

File: installer/windows/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+* `microk8s.exe`:  This is the file generated by PyInstaller.
+* `multipass.exe`:  This is the Multipass installer exe, which is run at install time.
+
+Use the following command to compile the installer.
+
+```
+makensis microk8s-windows.nsi
```
Comment (ktsakalozos) [inline installer/windows/README.md:21]: should this be `makensis microk8s/windows.nsi` ?

File: installer/windows/microk8s.nsi [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        ${NSD_CreateText} 42% 67.5 50u 10u "1.18/stable"
+        Pop $VmConfigureDialogTrack
+
+        ${NSD_CreateLabel} 8% 102.5 100% 10u "These are the minimum recommended parameters for the VM running ${PRODUCT_NAME}"
+        Pop $VmConfigureDialogFootnote
+
+        nsDialogs::Show
+    endLaunch:
+        Abort
+FunctionEnd
```
Comment (ktsakalozos) [inline installer/windows/microk8s.nsi:150]: Consider the following layout:
VM-conf-layout
```
Function "ConfigureVm"
    ClearErrors
    ReadRegStr $0 HKLM "Software\canonical\multipass" "InstallerRebootRequired"
    ${IfNot} ${Errors}  ; Error means hypervisor is available, so we can proceed.
        ${If} $0 == "true"
            MessageBox MB_ICONEXCLAMATION "Cannot configure the ${PRODUCT_NAME} VM as a reboot is required.  Please re-run this wizard after reboot to configure the VM."
            SetRebootFlag true
            Abort
        ${ElseIf} $0 == "false"
            MessageBox MB_ICONEXCLAMATION "Cannot configure the ${PRODUCT_NAME} VM as VirtualBox needs to be installed.  Please re-run this wizard after installing VirtualBox to configure the VM."
            Abort
        ${EndIf}
    ${EndIf}
    MessageBox MB_YESNO "Do you want to configure and launch the ${PRODUCT_NAME} VM now?" /SD IDYES IDNO endLaunch
        nsDialogs::Create 1018
        Pop $VmConfigureDialog

        ${NSD_CreateLabel} 0% 25u 100% 12u "Please configure the VM running MicroK8s"
        Pop $VmConfigureDialogFootnote
        ${NSD_AddStyle} $VmConfigureDialogFootnote ${SS_Center}


        ${NSD_CreateLabel} 30% 40u 20% 10u "CPU (cores): "
        Pop $VmConfigureDialogCpuLabel
        ${NSD_AddStyle} $VmConfigureDialogCpuLabel ${SS_Right}

        ${NSD_CreateNumber} 50% 40u 25u 10u "2"
        Pop $VmConfigureDialogCpu



        ${NSD_CreateLabel} 30% 55u 20% 10u "Mem (GB): "
        Pop $VmConfigureDialogMemLabel
        ${NSD_AddStyle} $VmConfigureDialogMemLabel ${SS_Right}

        ${NSD_CreateNumber} 50% 55u 25u 10u "4"
        Pop $VmConfigureDialogMem



        ${NSD_CreateLabel} 30% 70u 20% 10u "Disk (GB): "
        Pop $VmConfigureDialogDiskLabel
        ${NSD_AddStyle} $VmConfigureDialogDiskLabel ${SS_Right}

        ${NSD_CreateNumber} 50% 70u 25u 10u "50"
        Pop $VmConfigureDialogDisk



        ${NSD_CreateLabel} 30% 85u 20% 10u "Snap Track: "
        Pop $VmConfigureDialogTrackLabel
        ${NSD_AddStyle} $VmConfigureDialogTrackLabel ${SS_Right}

        ${NSD_CreateText} 50% 85u 85u 10u "1.18/stable"
        Pop $VmConfigureDialogTrack


        nsDialogs::Show
    endLaunch:
        Abort
FunctionEnd
```

File: installer/requirements.txt (hunk 1 above)
Comment (joedborg) [reply installer/requirements.txt:13]: I'll try it again, but it was doing something weird with GH actions.  You can always do `python3 ./setup.py install` to achieve that (it's what I've been doing).

**Pull request merged**